apps/dashboard/src/main/java/com/akto/action/observe/Utils.java

```diff
@@ -39,6 +40,39 @@ public static List<BasicDBObject> fetchEndpointsInCollectionUsingHost(int apiCol
         }
     }
 
+    public final static String _START_TS = "startTs";
+    public final static String _LAST_SEEN_TS = "lastSeenTs";
+
+    public static int countEndpoints(Bson filters) {
+        int ret = 0;
+
+        List<Bson> pipeline = new ArrayList<>();
+        BasicDBObject groupedId = new BasicDBObject("apiCollectionId", "$apiCollectionId")
+                .append("url", "$url")
+                .append("method", "$method");
+
+        Bson projections = Projections.fields(
+                Projections.include("timestamp", "lastSeen", "apiCollectionId", "url", "method"));
+
+        pipeline.add(Aggregates.project(projections));
+        pipeline.add(Aggregates.group(groupedId,
+                Accumulators.max(_START_TS, "$timestamp"),
+                Accumulators.max(_LAST_SEEN_TS, "$lastSeen")));
+        pipeline.add(Aggregates.match(filters));
```

Comment:
Match should come earlier (preferably first). This reduces query time by large amount

Reply:
@notshivansh ^^

Reply:
@avneesh-akto fixed.